java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/cdk/TearDownMavenSystemTest.java

```diff
@@ -74,11 +78,16 @@ public static void tearDown(
                 cloudFormation.deleteStack(builder -> builder.stackName(instanceId));
             } catch (RuntimeException e) {
                 LOGGER.warn("Failed deleting instance stack: " + instanceId, e);
+                tearDownFailures.add(new TearDownFailure(instanceId, e));
             }
         }
         for (String instanceId : instanceIds) {
             LOGGER.info("Waiting for instance CloudFormation stack to delete: {}", instanceId);
-            WaitForStackToDelete.from(cloudFormation, instanceId).pollUntilFinished();
+            try {
+                WaitForStackToDelete.from(cloudFormation, instanceId).pollUntilFinished();
+            } catch (DeleteFailedException e) {
+                tearDownFailures.add(new TearDownFailure(instanceId, e));
+            }
```

Comment:
This will cause it to carry on the tear down even if one instance fails. Is that correct? We probably don't want to tear down the system test cluster associated with an instance until we can tear down the instance?